From: Peymaneh Nejad <p.nejad@posteo.de>
Date: Fri, 12 Nov 2021 14:20:28 +0100
Subject: Remove dysfunct commands

These are dysfunct within a Debian-distributed Caddy-binary and removed:

* add-package
* build-info
* list-modules
* remove-package
* upgrade

Forwarded: not-needed
---
 cmd/commands.go | 69 ---------------------------------------------------------
 1 file changed, 69 deletions(-)

diff --git a/cmd/commands.go b/cmd/commands.go
index 1e2c40d..c00cd0f 100644
--- a/cmd/commands.go
+++ b/cmd/commands.go
@@ -199,26 +199,6 @@ config file; otherwise the default is assumed.`,
 		Short: "Prints the version",
 	})
 
-	RegisterCommand(Command{
-		Name:  "list-modules",
-		Func:  cmdListModules,
-		Usage: "[--packages] [--versions]",
-		Short: "Lists the installed Caddy modules",
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("list-modules", flag.ExitOnError)
-			fs.Bool("packages", false, "Print package paths")
-			fs.Bool("versions", false, "Print version information")
-			fs.Bool("skip-standard", false, "Skip printing standard modules")
-			return fs
-		}(),
-	})
-
-	RegisterCommand(Command{
-		Name:  "build-info",
-		Func:  cmdBuildInfo,
-		Short: "Prints information about this build",
-	})
-
 	RegisterCommand(Command{
 		Name:  "environ",
 		Func:  cmdEnviron,
@@ -288,55 +268,6 @@ is always printed to stdout.`,
 			return fs
 		}(),
 	})
-
-	RegisterCommand(Command{
-		Name:  "upgrade",
-		Func:  cmdUpgrade,
-		Short: "Upgrade Caddy (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the same modules/plugins at the
-latest versions. EXPERIMENTAL: May be changed or removed.`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("upgrade", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
-	RegisterCommand(Command{
-		Name:  "add-package",
-		Func:  cmdAddPackage,
-		Usage: "<packages...>",
-		Short: "Adds Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the specified packages (module/plugin)
-added. Retains existing packages. Returns an error if the any of packages are 
-already included. EXPERIMENTAL: May be changed or removed.
-`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("add-package", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
-	RegisterCommand(Command{
-		Name:  "remove-package",
-		Func:  cmdRemovePackage,
-		Usage: "<packages...>",
-		Short: "Removes Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binaries without the specified packages (module/plugin). 
-Returns an error if any of the packages are not included. 
-EXPERIMENTAL: May be changed or removed.
-`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("remove-package", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
 }
 
 // RegisterCommand registers the command cmd.
